File: src/o1tkeklargestnumber/Go/main.go
// File: main.go

package main

import (
	"fmt"
	"o1tkeklargestnumber/kthlargest"
	"strings"
)

func main() {
	nums := []int{3, 6, 9, 10}
	fmt.Print("Initial stream: ")
	printArray(nums)
	fmt.Println("\nk:", 3)
	kLargest := kthlargest.NewKthLargest(3, nums)
	val := []int{4, 7, 10, 8, 15}
	for _, v := range val {
		fmt.Println("\tAdding a new number", v, "to the stream")
		nums = append(nums, v)
		fmt.Print("\tNumber stream: ")
		printArray(nums)
		fmt.Println("\n\tKth largest element in the stream:", kLargest.Add(v))
		fmt.Println(strings.Repeat("-", 100))
	}
}

func printArray(arr []int) {
	fmt.Printf("[%v]", strings.Trim(strings.Join(strings.Fields(fmt.Sprint(arr)), ", "), "[]"))
}

File: src/o1tkeklargestnumber/Python/main.py
# File: main.py

from kth_largest import KthLargest


def print_array(arr):
    print("[{}]".format(", ".join(map(str, arr))))


def main():
    nums = [3, 6, 9, 10]
    print("Initial stream: ", end="")
    print_array(nums)
    print("\nk:", 3)
    k_largest = KthLargest(3, nums)
    values_to_add = [4, 7, 10, 8, 15]

    for val in values_to_add:
        print(f"\tAdding a new number {val} to the stream")
        nums.append(val)
        print("\tNumber stream: ", end="")
        print_array(nums)
        print(f"\n\tKth largest element in the stream: {k_largest.add(val)}")
        print("-" * 100)


if __name__ == "__main__":
    main()
